File: backend/core/branding.py
"""
Branding Configuration Module

Centralizes all branding-related configuration for easy client customization.
Modify these values to rebrand the application for your clients.
"""

import os

class BrandingConfig:
    """Application branding configuration"""
    
    # Application Name
    APP_NAME = os.getenv('APP_NAME', 'PythonWebApp')
    APP_SHORT_NAME = os.getenv('APP_SHORT_NAME', 'PWA')
    
    # Company Information
    COMPANY_NAME = os.getenv('COMPANY_NAME', 'Your Company')
    COMPANY_URL = os.getenv('COMPANY_URL', 'https://example.com')
    
    # API Documentation
    API_TITLE = os.getenv('API_TITLE', f'{APP_NAME} API')
    API_DESCRIPTION = os.getenv('API_DESCRIPTION', f'API Documentation for {APP_NAME}')
    API_VERSION = os.getenv('API_VERSION', '1.0.0')
    
    # Contact Information
    CONTACT_NAME = os.getenv('CONTACT_NAME', APP_NAME)
    CONTACT_URL = os.getenv('CONTACT_URL', 'https://github.com')
    CONTACT_EMAIL = os.getenv('CONTACT_EMAIL', 'contact@example.com')
    
    # Logo and Assets
    LOGO_PATH = os.getenv('LOGO_PATH', '/static/logo.png')
    FAVICON_PATH = os.getenv('FAVICON_PATH', '/static/favicon.ico')
    
    # Theme Colors (for future customization)
    PRIMARY_COLOR = os.getenv('PRIMARY_COLOR', '#0d6efd')
    SECONDARY_COLOR = os.getenv('SECONDARY_COLOR', '#6c757d')
    
    # Application Description
    APP_DESCRIPTION = os.getenv(
        'APP_DESCRIPTION',
        'Application moderne de gestion d\'utilisateurs avec React & Flask'
    )
    
    # Footer Copyright
    COPYRIGHT_YEAR = os.getenv('COPYRIGHT_YEAR', '2025')
    COPYRIGHT_HOLDER = os.getenv('COPYRIGHT_HOLDER', COMPANY_NAME)
    
    @classmethod
    def get_swagger_config(cls):
        """Get Swagger API documentation configuration"""
        return {
            "swagger": "2.0",
            "info": {
                "title": cls.API_TITLE,
                "description": cls.API_DESCRIPTION,
                "version": cls.API_VERSION,
                "contact": {
                    "name": cls.CONTACT_NAME,
                    "url": cls.CONTACT_URL,
                    "email": cls.CONTACT_EMAIL
                }
            }
        }
    
    @classmethod
    def get_app_info(cls):
        """Get application information for API responses"""
        return {
            "name": cls.APP_NAME,
            "version": cls.API_VERSION,
            "description": cls.APP_DESCRIPTION,
            "company": cls.COMPANY_NAME,
            "company_url": cls.COMPANY_URL
        }

File: frontend/src/config/branding.js
const BrandingConfig = {
  // Application Name
  appName: process.env.REACT_APP_NAME || 'PythonWebApp',
  appShortName: process.env.REACT_APP_SHORT_NAME || 'PWA',
  
  // Company Information
  companyName: process.env.REACT_APP_COMPANY_NAME || 'Your Company',
  companyUrl: process.env.REACT_APP_COMPANY_URL || 'https://example.com',
  
  // Application Description
  appDescription: process.env.REACT_APP_DESCRIPTION || 'Application moderne de gestion d\'utilisateurs avec React & Flask',
  appTagline: process.env.REACT_APP_TAGLINE || 'Bienvenue sur PythonWebApp',
  
  // Logo and Assets
  logoPath: process.env.REACT_APP_LOGO_PATH || '/logo.png',
  faviconPath: process.env.REACT_APP_FAVICON_PATH || '/favicon.ico',
  
  // Contact Information
  contactEmail: process.env.REACT_APP_CONTACT_EMAIL || 'contact@example.com',
  supportUrl: process.env.REACT_APP_SUPPORT_URL || '/support',
  
  // Social Media (optional)
  socialMedia: {
    twitter: process.env.REACT_APP_TWITTER || null,
    linkedin: process.env.REACT_APP_LINKEDIN || null,
    github: process.env.REACT_APP_GITHUB || null,
  },
  
  // Footer Copyright
  copyrightYear: process.env.REACT_APP_COPYRIGHT_YEAR || '2025',
  copyrightHolder: process.env.REACT_APP_COPYRIGHT_HOLDER || 'Your Company',
  
  // Theme Colors (for future customization)
  colors: {
    primary: process.env.REACT_APP_PRIMARY_COLOR || '#0d6efd',
    secondary: process.env.REACT_APP_SECONDARY_COLOR || '#6c757d',
  },
  
  // Features Toggles
  features: {
    darkMode: process.env.REACT_APP_FEATURE_DARK_MODE !== 'false',
    registration: process.env.REACT_APP_FEATURE_REGISTRATION !== 'false',
    apiDocs: process.env.REACT_APP_FEATURE_API_DOCS !== 'false',
  },
};

export default BrandingConfig;
